### ISSUE TITLE
Useless/Wrong GLSL code in depth test

### ISSUE THREAD

**Issue by SunBlack** [AnalyticalGraphicsInc/cesium#5621]
#5166 introduced depth test for billboards, points and labels. But the GLSL code makes no sense:

```
gl_Position.z = min(gl_Position.z, gl_Position.w);
bool clipped = gl_Position.z < -gl_Position.w || gl_Position.z > gl_Position.w;
```
This code I found in `BillboardCollectionVS.glsl` and `PointPrimitiveCollectionVS.glsl` (didn't found depth test code for labels now).

Case 1 (gl_Position.z < gl_Position.w):
```
gl_Position.z = min(0.0, 1.0); //gl_Position.z => 0.0, if gl_Position.z = 0.0 and gl_Position.w = 1.0
bool clipped = 0.0 < -1.0 || 0.0 > 1.0; //false
```

Case 2 (gl_Position.z > gl_Position.w):
```
gl_Position.z = min(1.0, 0.0); //gl_Position.z => 0.0, if gl_Position.z = 1.0 and gl_Position.w = 0.0
bool clipped = 0.0 < -0.0 || 0.0 > 0.0; //false
```

Case 3 (gl_Position.z == gl_Position.w):
```
gl_Position.z = min(0.0, 0.0); //gl_Position.z => 0.0, if gl_Position.z = 0.0 and gl_Position.w = 0.0
bool clipped = 0.0 < -0.0 || 0.0 > 0.0; //false
```

So `clipped` will be always `false`.

PS: Sorry, I have no idea how you handle code review if code is already merged ;).

**Comment by emackey** [comment]
This sounds like it could be the cause of #5501 and/or #5331.

/cc @rahwang

**Comment by emackey** [comment]
Thanks for pointing out this code, @SunBlack.  It's fixed in master, and will be released with 1.37.
